component/src/main/java/io/siddhi/extension/store/mongodb/MongoSelectExpressionVisitor.java

```diff
@@ -0,0 +1,307 @@
+                (!Arrays.asList(supportedFunctions).contains(functionName))) {
+            throw new MongoTableException("The MongoDB Event table does not support functions other than sum(), " +
+                    "avg(), min(), max() and count().");
+        }
+        if (functionName.equalsIgnoreCase("count")) {
+            this.isCountFunction = true;
+        }
+    }
+
+    @Override
+    public void endVisitAttributeFunction(String namespace, String functionName) {
+        if (MongoTableUtils.isEmpty(namespace) &&
+                (Arrays.asList(supportedFunctions).contains(functionName))) {
+            this.isAttributeFunctionUsed = true;
+            if (functionName.equalsIgnoreCase("count")) {
+                this.isCountFunction = false;
+                conditionOperands.push("{$sum:1}");
+            } else {
+                String functionArgument = conditionOperands.pop();
+                conditionOperands.push("{$" + functionName + ":" + functionArgument + "}");
+            }
+        }
+    }
+
+    @Override
+    public void beginVisitParameterAttributeFunction(int index) {
+    }
+
+    @Override
+    public void endVisitParameterAttributeFunction(int index) {
+    }
+
+    @Override
+    public void beginVisitMath(MathOperator mathOperator) {
+    }
+
+    @Override
+    public void endVisitMath(MathOperator mathOperator) {
+        String rightOperand = this.conditionOperands.pop();
+        String leftOperand = this.conditionOperands.pop();
+        conditionOperands.push("{$" + mathOperator.name().toLowerCase() + ":" + "[" + leftOperand + "," +
+                rightOperand + "]}");
+    }
+
+    @Override
+    public void beginVisitMathLeftOperand(MathOperator mathOperator) {
+    }
+
+    @Override
+    public void endVisitMathLeftOperand(MathOperator mathOperator) {
+    }
+
+    @Override
+    public void beginVisitMathRightOperand(MathOperator mathOperator) {
+    }
+
+    @Override
+    public void endVisitMathRightOperand(MathOperator mathOperator) {
+    }
+
+    @Override
+    public void beginVisitOr() {
+    }
+
+    @Override
+    public void endVisitOr() {
+        String rightOperand = this.conditionOperands.pop();
+        String leftOperand = this.conditionOperands.pop();
+        conditionOperands.push("{$or:" + "[" + leftOperand + "," + rightOperand + "]}");
+    }
+
+    @Override
+    public void beginVisitOrLeftOperand() {
+    }
+
+    @Override
+    public void endVisitOrLeftOperand() {
+    }
+
+    @Override
+    public void beginVisitOrRightOperand() {
+    }
+
+    @Override
+    public void endVisitOrRightOperand() {
+    }
+
+    @Override
+    public void beginVisitAnd() {
+    }
+
+    @Override
+    public void endVisitAnd() {
+        String rightOperand = this.conditionOperands.pop();
+        String leftOperand = this.conditionOperands.pop();
+        conditionOperands.push("{$and:" + "[" + leftOperand + "," + rightOperand + "]}");
```

Comment:
Same

Reply:
Fixed in 
https://github.com/siddhi-io/siddhi-store-mongodb/pull/56/commits/94fba768238670966474ed6ac8a757455790895b